src/blockfilter.cpp

```diff
@@ -208,7 +208,7 @@ static GCSFilter::ElementSet BasicFilterElements(const CBlock& block,
     for (const CTransactionRef& tx : block.vtx) {
         for (const CTxOut& txout : tx->vout) {
             const CScript& script = txout.scriptPubKey;
-            if (script[0] == OP_RETURN) continue;
+            if (script.size() > 0 && script[0] == OP_RETURN) continue;
             elements.emplace(script.begin(), script.end());
```

Comment:
Now it looks like it adds size 0 entries to the filter, which is a violation of BIP158.

I believe this should be "script.size() < 1 ||". No?

Reply:
nit to @gmaxwell comment, `script.empty() || ...`.